homeassistant/components/yamaha_musiccast/media_player.py

```diff
@@ -124,6 +137,56 @@ async def async_setup_entry(
 
     async_add_entities(media_players)
 
+    @service.verify_domain_control(hass, DOMAIN)
+    async def async_group_service_handle(service_call: ServiceCall):
+        """Handle services."""
+        entity_ids = service_call.data.get("entity_id", [])
+        if not entity_ids:
+            return
+
+        all_entities = []
+        for coord in hass.data[DOMAIN].values():
+            all_entities += coord.entities
+
+        entities = [entity for entity in all_entities if entity.entity_id in entity_ids]
+
+        if service_call.service == SERVICE_JOIN:
+            master_id = service_call.data[ATTR_MASTER]
+            master = next(
+                (entity for entity in all_entities if entity.entity_id == master_id),
+                None,
+            )
+            if master and isinstance(master, MusicCastDeviceEntity):
+                await master.async_server_join(entities)
+            else:
+                _LOGGER.error(
+                    "Invalid master specified for join service: %s",
+                    service_call.data[ATTR_MASTER],
+                )
+        elif service_call.service == SERVICE_UNJOIN:
+            for entity in entities:
+                if isinstance(entity, MusicCastDeviceEntity):
+                    await entity.async_unjoin()
+                else:
+                    _LOGGER.error(
+                        "Invalid entity specified for unjoin service: %s",
+                        entity,
+                    )
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_JOIN,
+        async_group_service_handle,
+        JOIN_SERVICE_SCHEMA,
+    )
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_UNJOIN,
+        async_group_service_handle,
+        UNJOIN_SERVICE_SCHEMA,
+    )
```

Comment:
Instead of creating your own services, please use the capabilities of the `media_player`, which already provide methods for this.'

See: <https://developers.home-assistant.io/docs/core/entity/media-player#grouping-player-entities-together>

Reply:
Okay, I did not know about that. Is there a way to create groups in the UI using these methods? The mini-media-player does not support these services, which were defined in the media_player. There is already a [task](https://github.com/kalkih/mini-media-player/issues/519) for this, but I would really love to support this card from the beginning.
Would it be okay to implement the methods from the media_player and add the domain specific services (as I did it) to support the mini-media-player?

Reply:
> Is there a way to create groups in the UI using these methods? 

They are exposed as services.

> The mini-media-player does not support these services

That is a third-party component, we do not support those and don't adjust our designs on those.

> Would it be okay to implement the methods from the media_player and add the domain specific services

No, please use the ones provided by the media_player.

Reply:
okay, done ✅